pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -70,6 +73,9 @@ public static TransformFunctionType getTransformFunctionType(String functionName
     try {
       return TransformFunctionType.valueOf(upperCaseFunctionName);
     } catch (Exception e) {
+      if (FunctionRegistry.getFunctionByName(functionName) != null) {
```

Comment:
This looks a bit hacky especially if FunctionRegistry starts getting used for something else as well (in addition to scalar functions). May be update the Javadoc of FunctionRegistry to state it is for scalar functions?

Currently it says in-built Pinot functions which is misleading since today all functions are in-built functions.

Reply:
@siddharthteotia Updated the doc, and also added a TODO to merge FunctionRegistry and FunctionDefinitionRegistry to provide one single registry for all functions.

Reply:
Awesome. Thanks!